metaserver/src/main/scala/scala/meta/languageserver/ScalametaLanguageServer.scala

```diff
@@ -200,13 +200,50 @@ class ScalametaLanguageServer(
   override def documentSymbols(
       td: TextDocumentIdentifier
   ): Seq[SymbolInformation] = {
-    val path = Uri.toPath(td.uri).get
-    symbol.documentSymbols(path.toRelative(cwd)).map {
-      case (position, denotation @ Denotation(_, name, signature, _)) =>
-        val location = path.toLocation(position)
-        val kind = denotation.symbolKind
-        SymbolInformation(name, kind, location, Some(signature))
+    import scala.meta._
+
+    // For a given node returns its closest ancestor which is a definition, declaration or a package object
+    // NOTE: package is not considered a wrapping definition, but it could be (a subject to discuss)
+    def wrappingDefinition(t: Tree): Option[Tree] = {
+      if (
+        t.is[Defn] ||
+        t.is[Decl] ||
+        t.is[Pkg.Object]
+      ) Some(t)
+      else t.parent.flatMap(wrappingDefinition)
+    }
+
+    // This is needed only to unfold full package names
+    def qualifiedName(t: Tree): Option[String] = t match {
+      case Term.Name(name) =>
+        Some(name)
+      case Term.Select(qual, name) =>
+        qualifiedName(qual).map { prefix => s"${prefix}.${name}" }
+      case Pkg(sel: Term.Select, _) =>
```

Comment:
What about `package a`?

Reply:
It falls into the `Member` case, so it will be just its `name` `a`. 
This whole method is here **only** for the packages with `Select`, because their default `name` is just the last part and I think it's useful to see the full name in the outline.

Reply:
I see, nice!